opus/application/test_api/test_cart_api.py

```diff
@@ -311,7 +311,7 @@ def test__api_cart_add_download_one(self):
_version': 'moon_geometry@Current'}, {'slug_name': 'ring_geometry', 'tooltip': 'Text files ([volume]_ring_summary.tab) that list the values of various ring plane intercept geometry metadata for every observation in a particular volume. Associated labels (*.lbl) describe the contents of the text files.', 'product_type': 'Ring Geometry Index', 'product_count': 1, 'download_count': 2, 'download_size': 2229982, 'download_size_pretty': '2M', 'default_checked': 0, 'product_type_with_version': 'ring_geometry@Current'}]}, 'Browse Products': {'Current': [{'slug_name': 'browse_thumb', 'tooltip': 'Thumbnail-size (often 100x100) non-linearly stretched preview JPEGs (*_thumb.jpg) of observations created by the Ring-Moon Systems Node. Previews of images are colored according to the filter used. Previews from non-imaging instruments attempt to represent the contents of observations in a visual way. Previews are not appropriate for scientific use.', 'product_type': 'Browse Image (thumbnail)', 'product_count': 1, 'download_count': 1, 'download_size': 995, 'download_size_pretty': '995B', 'default_checked': 0, 'product_type_with_version': 'browse_thumb@Current'}, {'slug_name': 'browse_small', 'tooltip': 'Small-size (often 256x256) non-linearly stretched preview JPEGs (*_small.jpg) of observations created by the Ring-Moon Systems Node. Previews of images are colored according to the filter used. Previews from non-imaging instruments attempt to represent the contents of observations in a visual way. Previews are not appropriate for scientific use.', 'product_type': 'Browse Image (small)', 'product_count': 1, 'download_count': 1, 'download_size': 3012, 'download_size_pretty': '2K', 'default_checked': 0, 'product_type_with_version': 'browse_small@Current'}, {'slug_name': 'browse_medium', 'tooltip': 'Medium-size (often 512x512) non-linearly stretched preview JPEGs (*_med.jpg) of observations created by the Ring-Moon Systems Node. Previews of images are colored according to the filter used. Previews from non-imaging instruments attempt to represent the contents of observations in a visual way. Previews are not appropriate for scientific use.', 'product_type': 'Browse Image (medium)', 'product_count': 1, 'download_count': 1, 'download_size': 7892, 'download_size_pretty': '7K', 'default_checked': 0, 'product_type_with_version': 'browse_medium@Current'}, {'slug_name': 'browse_full', 'tooltip': 'Full-size non-linearly stretched preview JPEGs (*_full.jpg) of observations created by the Ring-Moon Systems Node. Previews of images are not colored. Previews from non-imaging instruments attempt to represent the contents of observations in a visual way. Previews are not appropriate for scientific use.', 'product_type': 'Browse Image (full)', 'product_count': 1, 'download_count': 1, 'download_size': 215079, 'download_size_pretty': '210K', 'default_checked': 1, 'product_type_with_version': 'browse_full@Current'}]}}, 'error': False, 'count': 1, 'recycled_count': 0, 'reqno': 101010101}
```

Comment:
I just noticed there's an entire class of tests missing that you could add now. All of the cart operators like add, remove, and reset take an optional types= parameter when download=1 is specified, but we never test this. This would be a good time to add a few tests that add that parameter and also check versions at the same time.